File: week06/20/Pr3.c

//  * Definition for a binary tree node.
  struct TreeNode {
      int val;
      struct TreeNode *left;
      struct TreeNode *right;
  };

int sumOfLeftLeaves(struct TreeNode* root) {
    // Recursive Solution
    // Base case
    if(!root) return 0;
    int ans = 0;
    if(root->left && !root->left->left && !root->left->right){
        ans += root->left->val;
    }
    ans += sumOfLeftLeaves(root->right);
    ans += sumOfLeftLeaves(root->left);

    return ans;
}

File: week06/20/Pr3.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        ans = [0]
        if not root or not (root.left or root.right) : return ans[0]

        def DFS(node):
            if node.left:
                if not node.left.left and not node.left.right:
                    ans[0] += node.left.val
                DFS(node.left)
            if node.right:
                DFS(node.right)
        DFS(root)
        return ans[0]
        
            